File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        
        int left = 0;
        
        int right = height.size() - 1;
        
        int result = 0;
        
        while(left < right) {
            int area = (right - left) * min(height[left], height[right]);
            
            if (area > result) {
                result = area;
            }
            if (height[left] < height[right]) {
                left++;
            }
            else {
                right--;
            }
        }
        return result;
        
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        left = 0
        right = len(height) - 1
        result = 0
        
        while left < right:
            area = (right-left) * min(height[left], height[right])
            result = area if area > result else result
            
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        
        return result